### PR TITLE
Add FIPS-compliant build of openssl

### PR DESCRIPTION
**Pull request by amcmanus**
A FIPS-compliant build of openssl is required as part of VVSG2.0. This PR creates the playbook to install the latest version of openssl (3.0.9) that meets FIPS requirements. It also includes the necessary openssl.cnf to enable FIPS.

During the online phase, the openssl 3.0.9 source code is downloaded.
During the offline phase, openssl is compiled with FIPS enabled, and the config file and FIPS module are copied to the correct system locations. 

NOTE: In the future, we may want to support multiple versions, so that can be specified in inventory definitions but defaults to 3.0.9 for now. Adding a version would also require adding the appropriate openssl.cnf for that version.

### CODE REVIEW

File: inventories/v4.0.0/group_vars/all/packages.yaml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 ---
 batch_package_install: true
-apt_snapshot_date: "20241030"
-release_name: "bookworm"
+  #apt_snapshot_date: "20241030"
+  #release_name: "bookworm"
```
Comment (arsalansufi) [inline inventories/v4.0.0/group_vars/all/packages.yaml:4]: I know that this is largely unrelated to this change, but is there a new snapshot I should use for the RC1s that I prep today? Hoping to start locking things in

**Review by arsalansufi — APPROVED** [review]
LGTM! Thank you again for all the thought partnership on this!

File: inventories/v4.0.0/group_vars/all/packages.yaml (hunk 1 above)
Comment (amcmanus) [reply inventories/v4.0.0/group_vars/all/packages.yaml:4]: @arsalansufi, I don't think we've had any changes that required package updates, so you could go ahead and use that existing one. Keep in mind, a new base image will have to be created so it's only referencing that apt repo.

Comment (arsalansufi) [reply inventories/v4.0.0/group_vars/all/packages.yaml:4]: Okay great! In that case, let's maybe bring these lines back, and I'll make a new base image when I make images this evening

Comment (amcmanus) [reply inventories/v4.0.0/group_vars/all/packages.yaml]: Updated.

**Pull request merged**